lib/upgradable-multisig/multisig/validation.ak

```diff
@@ -0,0 +1,174 @@
+use aiken/list
+use aiken/transaction.{Output, OutputReference, ScriptContext, Transaction}
+use aiken/transaction/credential
+use aiken/transaction/value.{PolicyId, from_minted_value}
+use aiken_design_patterns/singular_utxo_indexer as singular_utxo_indexer
+use assist/data
+use assist/values
+use upgradable_multisig/multisig/utils.{
+  get_asset_amount, signed_within_threshold, validate_multisig_datum,
+  validate_token_output,
+}
+use upgradable_multisig/types.{MultisigDatum}
+
+pub fn validate_initiate_multisig(
+  own_policy: PolicyId,
+  output_reference: OutputReference,
+  input_index: Int,
+  tx: Transaction,
+) -> Bool {
+  let minted_tokens = tx.mint |> from_minted_value()
+  expect Some(init_input) = list.at(tx.inputs, input_index)
+  let multisig_credential = credential.ScriptCredential(own_policy)
+  // Ensure this outref is spent in this transaction
+  let is_outref_spent = init_input.output_reference == output_reference
+
+  expect [multisig_output, ..] = tx.outputs
```

Comment:
Let's not rely on positional inputs or outputs... makes the validator much less flexible. We should find the relevant output based the nft in its value.

Reply:
Alternatively we could use an indexer coming from the redeemer.